src/scenekit.cs

```diff
@@ -443,9 +467,196 @@ interface SCNCamera : SCNAnimatable, SCNTechniqueSupport, NSCopying, NSSecureCod
+	
+		[Watch (4, 0), TV (11, 0), Mac (10, 13), iOS (11, 0)]
+		[Export ("focusDistance")]
+		nfloat FocusDistance { get; set; }
+	
+		[Watch (4, 0), TV (11, 0), Mac (10, 13), iOS (11, 0)]
+		[Export ("focalBlurSampleCount")]
+		nint FocalBlurSampleCount { get; set; }
+
+		[Watch (4, 0), TV (11, 0), Mac (10, 13), iOS (11, 0)]
+		[Export ("fStop")]
+		nfloat FStop { get; set; }
+	
+		[Watch (4, 0), TV (11, 0), Mac (10, 13), iOS (11, 0)]
+		[Export ("apertureBladeCount")]
+		nint ApertureBladeCount { get; set; }
+	
+		[Watch (4, 0), TV (11, 0), Mac (10, 13), iOS (11, 0)]
+		[Export ("screenSpaceAmbientOcclusionIntensity")]
+		nfloat ScreenSpaceAmbientOcclusionIntensity { get; set; }
+	
+		[Watch (4, 0), TV (11, 0), Mac (10, 13), iOS (11, 0)]
+		[Export ("screenSpaceAmbientOcclusionRadius")]
+		nfloat ScreenSpaceAmbientOcclusionRadius { get; set; }
+	
+		[Watch (4, 0), TV (11, 0), Mac (10, 13), iOS (11, 0)]
+		[Export ("screenSpaceAmbientOcclusionBias")]
+		nfloat ScreenSpaceAmbientOcclusionBias { get; set; }
+	
+		[Watch (4, 0), TV (11, 0), Mac (10, 13), iOS (11, 0)]
+		[Export ("screenSpaceAmbientOcclusionDepthThreshold")]
+		nfloat ScreenSpaceAmbientOcclusionDepthThreshold { get; set; }
+	
+		[Watch (4, 0), TV (11, 0), Mac (10, 13), iOS (11, 0)]
+		[Export ("screenSpaceAmbientOcclusionNormalThreshold")]
+		nfloat ScreenSpaceAmbientOcclusionNormalThreshold { get; set; }
+
 #endif
 	}
 
+	[Watch (4,0), TV (11,0), Mac (10,13), iOS (11,0)]
+	[Protocol]
+	[BaseType (typeof(NSObject))]
+	interface SCNCameraControlConfiguration
+	{
+		[Abstract]
+		[Export ("autoSwitchToFreeCamera")]
+		bool AutoSwitchToFreeCamera { get; set; }
+	
+		[Abstract]
+		[Export ("allowsTranslation")]
+		bool AllowsTranslation { get; set; }
+	
+		[Abstract]
+		[Export ("flyModeVelocity")]
+		nfloat FlyModeVelocity { get; set; }
+	
+		[Abstract]
+		[Export ("panSensitivity")]
+		nfloat PanSensitivity { get; set; }
+	
+		[Abstract]
+		[Export ("truckSensitivity")]
+		nfloat TruckSensitivity { get; set; }
+	
+		[Abstract]
+		[Export ("rotationSensitivity")]
+		nfloat RotationSensitivity { get; set; }
+	}
+
+	interface ISCNCameraControllerDelegate {}
+	
+	[Watch (4,0), TV (11,0), Mac (10,13), iOS (11,0)]
+	[Protocol]
+	[Model] // Figured I would keep the model for convenience, as all the methods here are optional
+	[BaseType (typeof(NSObject))]
+	interface SCNCameraControllerDelegate
+	{
+		[Export ("cameraInertiaWillStartForController:")]
+		void CameraInertiaWillStart (SCNCameraController cameraController);
+	
+		[Export ("cameraInertiaDidEndForController:")]
+		void CameraInertiaDidEnd (SCNCameraController cameraController);
+	}
+	
+	[Watch (4,0), TV (11,0), Mac (10,13), iOS (11,0)]
+	[BaseType (typeof(NSObject))]
+	interface SCNCameraController
+	{
+		[Wrap ("WeakDelegate")]
+		[NullAllowed]
+		[Protocolize]
+		SCNCameraControllerDelegate Delegate { get; set; }
+	
+		[NullAllowed, Export ("delegate", ArgumentSemantic.Assign)]
+		NSObject WeakDelegate { get; set; }
```

Comment:
That does look like a bad idea.

I looked for examples of things that did this, and what it means is that now there is no way for users to get compiler-verified implementations of the delegate classes at all.  While this might be acceptable in scenarios where the [Protocol] labeled class has all of its members be Abstract, that does not work for optional methods, where the alternative provides absolutely no benefit.

I tried looking for alternatives in the generated code and did not see it, perhaps I am missing something, but if the idea is to only offer a weakly typed interface, that is not acceptable, and we need to revisit every scenario where this became a policy and fix it to work in this way.